consensus/viewchange/roster/mod.go

```diff
@@ -304,3 +384,35 @@ func (f defaultFactory) FromProto(in proto.Message) (viewchange.Authority, error
 
 	return roster, nil
 }
+
+// VisitJSON implements serde.Factory. It deserializes the roster in JSON
+// format.
+func (f defaultFactory) VisitJSON(in serde.FactoryInput) (serde.Message, error) {
+	m := json.Roster{}
+	err := in.Feed(&m)
+	if err != nil {
+		return nil, xerrors.Errorf("couldn't deserialize roster: %v", err)
+	}
+
+	addrs := make([]mino.Address, len(m))
+	pubkeys := make([]crypto.PublicKey, len(m))
+
+	for i, raw := range m {
```

Comment:
would find it clearer
```suggestion
	for i, player := range m {
```